gitlab/__init__.py

```diff
@@ -397,16 +397,22 @@ def _raw_get(self, path_, content_type=None, streamed=False, **kwargs):
             url = '%s%s' % (self._url, path_)
 
         headers = self._create_headers(content_type)
+
+        if 'Authorization' in self.headers:
+            auth = None
+        else:
+            auth = requests.auth.HTTPBasicAuth(
+                self.http_username,
+                self.http_password)
+
```

Comment:
There is one thing bothering me here: with oauth token we explicitly define headers, with basic HTTP auth we don't and use a python-requests feature instead. I'd prefer to have the same method for both authentication methods.

Updating the ``_create_headers`` method might be the easiest way to achieve consistency. This would also avoid code duplication in the ``_raw_*`` methods.

What do you think about this?